### PR TITLE
DID wallet coin_added

### PR DESCRIPTION
**Pull request by ytx1991**
### Purpose:
Prevent redundant peer call in the DID wallet.

### Current Behavior:
DID will call peer API again to get coin state and coin spend.

### New Behavior:
Reuse the coin state and coin spend got in the wallet manager.

### CODE REVIEW

File: tests/wallet/did_wallet/test_did.py [hunk 1]
```diff
@@ -793,10 +793,8 @@ async def test_did_transfer(self, self_hostname, two_wallet_nodes, with_recovery
         assert metadata["GitHub"] == "测试"
 
         # Test match_hinted_coin
-        assert await did_wallet_2.match_hinted_coin(
-            await did_wallet_2.get_coin(),
-            new_puzhash,
-        )
+        if with_recovery:
+            assert await did_wallet_2.match_hinted_coin(await did_wallet_2.get_coin(), new_puzhash)
```
Comment (Quexington) [inline tests/wallet/did_wallet/test_did.py:797]: Why this change?

Comment (ytx1991) [reply tests/wallet/did_wallet/test_did.py:797]: In the without recovery info mode, the recovery setting will be reset so the match will fail.

Comment (Quexington) [reply tests/wallet/did_wallet/test_did.py:797]: But why did it break in this PR?  This is supposed to be an optimization right so why did the behavior change?

Comment (ytx1991) [reply tests/wallet/did_wallet/test_did.py]: I thought there is a bug, revert the change.

File: chia/wallet/did_wallet/did_wallet.py [hunk 2]
```diff
@@ -376,7 +394,7 @@ async def coin_added(self, coin: Coin, _: uint32, peer: WSChiaConnection, coin_d
             assert full_puzzle.get_tree_hash() == coin.puzzle_hash
         if self.did_info.temp_coin is not None:
             self.wallet_state_manager.state_changed("did_coin_added", self.wallet_info.id)
-
+        print(f"wallet {self.get_name()} {did_data.num_verification} {self.did_info.num_of_backup_ids_needed}")
```
Comment (Quexington) [inline chia/wallet/did_wallet/did_wallet.py:397]: Is this a leftover debug line?

**Review by Quexington — DISMISSED** [review]

**Review by xdustinface — CHANGES_REQUESTED** [review]
Seems like the last catch up merge messed something up here and there is one unrelated formatting change.

**Review by xdustinface — COMMENTED** [review]
pre-commit (`isort`) is still broken.

**Review by xdustinface — APPROVED** [review]

**Review by wjblanke — APPROVED** [review]
aok
